config/configgrpc/configgrpc_test.go

```diff
@@ -963,6 +963,118 @@ func TestClientInfoInterceptors(t *testing.T) {
 	}
 }
 
+func TestClientMetadataWithAuthInterceptorsAndIncludeMetadata(t *testing.T) {
+	testCases := []struct {
+		desc            string
+		tester          func(context.Context, ptraceotlp.GRPCClient)
+		includeMetadata bool
+	}{
+		{
+			desc: "overwritten",
+			tester: func(ctx context.Context, cl ptraceotlp.GRPCClient) {
+				resp, errResp := cl.Export(ctx, ptraceotlp.NewExportRequest())
+				require.NoError(t, errResp)
+				require.NotNil(t, resp)
+			},
+			includeMetadata: true,
+		},
+		{
+			desc: "found",
+			tester: func(ctx context.Context, cl ptraceotlp.GRPCClient) {
+				resp, errResp := cl.Export(ctx, ptraceotlp.NewExportRequest())
+				require.NoError(t, errResp)
+				require.NotNil(t, resp)
+			},
+			includeMetadata: false,
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			mock := &grpcTraceServer{}
+			var l net.Listener
+
+			// prepare the server
+			{
+				gss := &ServerConfig{
+					NetAddr: confignet.AddrConfig{
+						Endpoint:  "localhost:0",
+						Transport: confignet.TransportTypeTCP,
+					},
+					IncludeMetadata: tC.includeMetadata,
+				}
+				gss.Auth = &configauth.Authentication{
+					AuthenticatorID: mockID,
+				}
+				host := &mockHost{
+					ext: map[component.ID]component.Component{
+						mockID: auth.NewServer(auth.WithServerAuthenticate(func(ctx context.Context, headers map[string][]string) (context.Context, error) {
+							return client.NewContext(ctx, client.Info{
```

Comment:
The right thing here would be to obtain the client from the context, set _only_ the Auth attributes, and wrap it in the context that is being returned, like this:

```go
mockID: auth.NewServer(auth.WithServerAuthenticate(func(ctx context.Context, headers map[string][]string) (context.Context, error) {
	cl := client.FromContext(ctx)
	cl.Auth = &mockAuthData{
		Attributes: map[string]string{"some-key-set-in-auth": "some-value-set-in-auth"},
	}

	return client.NewContext(ctx, cl), nil
})),
```